### PR TITLE
Orchestration core: data model, resolver

### PR DESCRIPTION
**Pull request by ebensom**
**Description**

Part of "Orchestration of SKR changes" epic. 

Changes proposed in this pull request:

- orchestration core: data model
- orchestration core: RuntimeResolver implementation

### CODE REVIEW

File: components/kyma-environment-broker/internal/orchestration/model.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	Resolve(include []RuntimeTarget, exclude []RuntimeTarget) ([]Runtime, error)
+}
+
+// OrchestrationStrategy interface encapsulates the strategy how the orchestration is performed.
+type OrchestrationStrategy interface {
+	// Execute invokes operation managers' Execute(operationID string) method for each operation according to the encapsulated strategy.
+	Execute(operations []RuntimeOperation) (time.Duration, error)
+}
+
+type OrchestrationStrategyType string
```
Comment (polskikiel) [inline components/kyma-environment-broker/internal/orchestration/model.go:55]: IDE complains about the naming - default usage of the types which name starts with `Orchestration` from external package will look like it: ```orchestration.OrchestrationStrategyType```
Can we just remove that Orchestration prefix from the structs names?

File: components/kyma-environment-broker/internal/orchestration/model.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+// Runtime is the data type which captures the needed SKR specific attributes to perform reconciliations on a given runtime.
+type Runtime struct {
+	InstanceID      string `json:"instanceId"`
+	RuntimeID       string `json:"runtimeId"`
+	GlobalAccountID string `json:"globalAccountId"`
+	SubAccountID    string `json:"subaccountId"`
+	// The corresponding shoot cluster's .metadata.name value
+	ShootName string `json:"shootName"`
+	// The corresponding shoot cluster's .spec.maintenance.timeWindow.Begin value, in "HHMMSS+[TZ HHMM]" format, e.g. "040000+0000"
+	MaintenanceWindowBegin string `json:"maintenanceWindowBegin"`
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/orchestration/model.go:16]: the time as string seems to be useless, I think it is better to use time.Time

File: components/kyma-environment-broker/internal/orchestration/resolver_test.go [hunk 3]
```diff
@@ -0,0 +1,547 @@
...
+	runtime3ID              = "7ba68c2a-0f63-4941-967d-30a9d3c30c0a"
+	runtime3InstanceID      = "a115079e-3ea4-4a60-8ec8-26f3b9d16583"
+
+	runtime4GlobalAccountID = "51b8c950-7ce3-4382-8670-99ae9549eabf"
+	runtime4SubAccountID    = "74cb2f93-4a30-4ff9-9cd0-4242f2d8227d"
+	runtime4ID              = "2cc9f953-5979-4efd-a2cd-ca60a125bf4a"
+	runtime4InstanceID      = "f89cfee8-2506-472c-ae92-d00697672917"
+)
+
+var shoot1 = gardenerapi.Shoot{
```
[the remark this answers is not in the document]
Comment (ebensom) [reply components/kyma-environment-broker/internal/orchestration/resolver_test.go:48]: Yea, I was also thinking about this. I will cover the normal scenarios with fixtures, and will leave static definitions for the corner cases.

File: components/kyma-environment-broker/internal/orchestration/model.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+// OrchestrationStrategySpec is the strategy part common for all orchestration trigger/status API
+type OrchestrationStrategySpec struct {
+	Type     OrchestrationStrategyType `json:"type"`
+	Schedule OrchestrationScheduleType `json:"schedule,omitempty"`
+	Parallel ParallelOrchestrationSpec `json:"parallel,omitempty"`
+}
+
+// TargetSpec is the targets part common for all orchestration trigger/status API
+type TargetSpec struct {
```
[the remark this answers is not in the document]
Comment (ebensom) [reply components/kyma-environment-broker/internal/orchestration/model.go:82]: Yes, I will use that in RuntimeResolver's function. The other use case will be that this struct is doing to be the targets: part of the orchestration API (and maybe the status response if we want to include the detail about the targets the specific orchestration was triggered with).

**Review by polskikiel — APPROVED** [review]

**Pull request merged**